client/components/navigation-header/navigation-header.tsx

```diff
@@ -83,7 +83,7 @@ const NavigationHeader: React.FC< HeaderProps > = ( {
 			} ) }
 			{ ...rest }
 		>
-			<div className="calypso-navigation-header__head">{ headElement }</div>
+			{ headElement && <div className="calypso-navigation-header__head">{ headElement }</div> }
```

Comment:
Don't add an extra class if we don't have any content to be displayed inside it. This also makes sure extra spacing is not added where this class is not displayed.